course3/homework-list/list.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <iterator>
+#include <memory>
+#include <initializer_list>
+#include <boost/range/adaptor/reversed.hpp>
+
+
+namespace Karun
+{
+
+template<typename T>
+class List
+{
+public:
+	typedef T value_type;
```

Comment:
This was required in order for std::front_inserter to work. Is there a better way?